src/ethereum/EnigmaContractWriterAPI.js

```diff
@@ -8,8 +8,12 @@ const config = require('./config.json');
 const EMPTY_HEX_STRING = '0x'; // This is the right value to pass an empty value to the contract, otherwise we get an error
 
 class EnigmaContractWriterAPI extends EnigmaContractReaderAPI {
-  constructor(enigmaContractAddress, enigmaContractABI, web3, logger) {
+  constructor(enigmaContractAddress, enigmaContractABI, web3, logger, walletAddress) {
     super(enigmaContractAddress, enigmaContractABI, web3, logger);
+    this._defaultTrxOptions = config.default;
+    if (walletAddress !== undefined && walletAddress !== null) {
```

Comment:
shouldn't you just do `if (walletAddress)`?

Reply:
Fixed